spark-operator/src/main/java/org/apache/spark/k8s/operator/metrics/sink/PrometheusPullModelSink.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.metrics.sink;
+
+import static org.apache.spark.k8s.operator.utils.ProbeUtil.sendMessage;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import com.codahale.metrics.MetricRegistry;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.spark.metrics.sink.PrometheusServlet;
+
+/** A simple Prometheus sink (pull model) implementation that presents metrics snapshot on demand */
+@Slf4j
+public class PrometheusPullModelSink extends PrometheusServlet implements HttpHandler {
```

Comment:
I'm not sure if we need to have `PrometheusPullModelSink` here, @jiangzho .

Reply:
I was in split mind as well - we need one class here despite `PrometheusPullModelSink` merely extends PrometheusServlet`. It also implements `HttpHandler` to be easily attached to a Http Server - and that would serve operator, which does not depend on netty and does not have an existing Spark UI. 

As the major difference between the two is the ability to filter empty results, I can update the other PR to introduce `getMetricsSnapshot(filterEmptyRecords: Boolean)` in `PrometheusServlet` in order to achieve minimal duplication at both side. This class can then be renamed as `PrometheusPullModelHandler` to avoid confusion, and can be lightweight, implementing only `handle`. How does this sound?